### PR TITLE
[in_app_purchase_storekit] Migrate FIATransactionCacheTests.m to.Swift

### PR DESCRIPTION
**Pull request by LouiseHsu**
Part of https://github.com/flutter/flutter/issues/151624

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I [linked to at least one issue that this PR fixes] in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style], or this PR is [exempt from CHANGELOG changes].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

### CODE REVIEW

**Comment by LouiseHsu** [comment]
Theres only 2 more test files after this 🙏

File: packages/in_app_purchase/in_app_purchase_storekit/example/shared/RunnerTests/FIATransactionCacheTests.swift [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+// found in the LICENSE file.
+
+import XCTest
+
+@testable import in_app_purchase_storekit
+
+final class FIATransactionCacheTests: XCTestCase {
+
+  func testAddObjectsForNewKey() {
+    let dummyArray: [Int] = [1, 2, 3]
```
Comment (hellohuanlin) [inline packages/in_app_purchase/in_app_purchase_storekit/example/shared/RunnerTests/FIATransactionCacheTests.swift:12]:
``` 
   let dummyArray = [1, 2, 3]
```

File: packages/in_app_purchase/in_app_purchase_storekit/example/shared/RunnerTests/FIATransactionCacheTests.swift [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    let cache = FIATransactionCache()
+    cache.add(dummyArray, for: TransactionCacheKey.updatedTransactions)
+
+    XCTAssertEqual(
+      dummyArray, cache.getObjectsFor(TransactionCacheKey.updatedTransactions) as! [Int])
```
Comment (hellohuanlin) [inline packages/in_app_purchase/in_app_purchase_storekit/example/shared/RunnerTests/FIATransactionCacheTests.swift:17]: can you use that unwrap function and avoid `!` here? 

```
let updatedTransactions = try XCTUnwrap(cache.getObjectsFor(TransactionCacheKey.updatedTransactions) as? [Int])
```

File: packages/in_app_purchase/in_app_purchase_storekit/example/shared/RunnerTests/FIATransactionCacheTests.swift [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+  func testGetObjectsForNonExistingKey() {
+    let cache = FIATransactionCache()
+    XCTAssert(cache.getObjectsFor(TransactionCacheKey.updatedTransactions).count == 0)
+  }
+
+  func testClear() {
+    let fakeUpdatedTransactions: [Int] = [1, 2, 3]
+    let fakeRemovedTransactions: [String] = ["Remove 1", "Remove 2", "Remove 3"]
+    let fakeUpdatedDownloads: [String] = ["Download 1", "Download 2"]
```
Comment (hellohuanlin) [inline packages/in_app_purchase/in_app_purchase_storekit/example/shared/RunnerTests/FIATransactionCacheTests.swift:42]: no need the type

**Review by hellohuanlin — APPROVED** [review]
some nits

File: packages/in_app_purchase/in_app_purchase_storekit/example/shared/RunnerTests/FIATransactionCacheTests.swift [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+
+    cache.add([4, 5, 6], for: TransactionCacheKey.updatedTransactions)
+
+    let expected: [Int] = [1, 2, 3, 4, 5, 6]
+    XCTAssertEqual(expected, cache.getObjectsFor(TransactionCacheKey.updatedTransactions) as! [Int])
+  }
+
+  func testGetObjectsForNonExistingKey() {
+    let cache = FIATransactionCache()
+    XCTAssert(cache.getObjectsFor(TransactionCacheKey.updatedTransactions).count == 0)
```
Comment (hellohuanlin) [inline packages/in_app_purchase/in_app_purchase_storekit/example/shared/RunnerTests/FIATransactionCacheTests.swift:36]: can be `.isEmpty`

File: packages/in_app_purchase/in_app_purchase_storekit/example/shared/RunnerTests/FIATransactionCacheTests.swift (hunk 2 above)
Comment (LouiseHsu) [reply packages/in_app_purchase/in_app_purchase_storekit/example/shared/RunnerTests/FIATransactionCacheTests.swift]: ooh yes, done

File: packages/in_app_purchase/in_app_purchase_storekit/example/shared/RunnerTests/FIATransactionCacheTests.swift (hunk 3 above)
Comment (LouiseHsu) [reply packages/in_app_purchase/in_app_purchase_storekit/example/shared/RunnerTests/FIATransactionCacheTests.swift]: removed all types

**Pull request merged**